utils/python/CIME/SystemTests/erp.py

```diff
@@ -11,134 +11,82 @@
-                expect(stop_new > 0, "ERROR: stop_n value %d too short %d %d"%(stop_new,stop_n,rest_n))
-                self._case.set_value("STOP_N", stop_new)
-                self._case.set_value("CONTINUE_RUN", True)
-                self._case.set_value("REST_OPTION","never")
-                suffix = "rest"
-            self.run_indv(suffix=suffix)
+        SystemTestsCompareTwo.__init__(self, case,
+                                       separate_builds = True,
+                                       run_two_suffix = "BFBRestTest",
+                                       run_one_st_archive = True,
+                                       run_one_description = "Baseline run from day 0 to STOP_N",
+                                       run_two_description = ("Test run starting from run one's " +
+                                                              "state at STOP_N // 2 + 1 to STOP_N.\n" +
+                                                              "Run 2 is done with half of the " +
+                                                              "number of tasks and threads as run 1"))
+
+    def _case_one_setup(self):
+        stop_n = self._case.get_value("STOP_N")
+        rest_n = _compute_rest_n(stop_n)
+        expect(stop_n > rest_n , "STOP_N value too small for test")
+        self._case.set_value("REST_N", rest_n)
+        self._case.set_value("REST_OPTION", self._case.get_value("STOP_OPTION"))
+
+    def _case_two_setup(self):
+        """ Case two uses half the number of threads and tasks as the defaults and case one. """
+        self._case.set_value("REST_OPTION", "never")
+
+        stop_n_1 = self._case1.get_value("STOP_N")
+        stop_n = stop_n_1 - _compute_rest_n(stop_n_1)
+        expect(stop_n > 0, "STOP_N value too small for test")
+        self._case.set_value("STOP_N", stop_n)
+
+        self._case.set_value("CONTINUE_RUN", True)
+
+        for comp in self._case.get_values("COMP_CLASSES"):
+            if comp == "DRV":
+                comp = "CPL"
+            ntasks = self._case.get_value("NTASKS_%s"%comp)
+            if ntasks > 1:
+                rootpe = self._case.get_value("ROOTPE_%s"%comp)
+                self._case.set_value("NTASKS_%s"%comp, ntasks // 2)
+                self._case.set_value("ROOTPE_%s"%comp, rootpe // 2)
+            nthreads = self._case.get_value("NTHRDS_%s"%comp)
+            if nthreads > 1:
+                self._case.set_value("BUILD_THREADED", True)
+                self._case.set_value("NTHRDS_%s"%comp, nthreads // 2)
+        case_setup(self._case, test_mode=True, reset=True)
+
+    def _inter_run_hook(self):
+        st_archive_dir_one = self._case1.get_value("DOUT_S_ROOT")
+        restdir_name = "rest"
+        restdir = os.path.join(st_archive_dir_one, restdir_name)
+        rundir = self._case2.get_value("RUNDIR")
+        path_1 = None
+        for root, _, files in os.walk(restdir):
+            for f in files:
+                if path_1 == None:
```

Comment:
Check for None with `is None` (see https://www.python.org/dev/peps/pep-0008/#programming-recommendations)

More importantly: Can you describe what you're doing with the logic here for setting path_1? This is unclear to me.